src/application/services/basic-folder-change-detection.service.ts

```diff
@@ -0,0 +1,73 @@
+import { RomachRepositoryInterface } from '../interfaces/romach-repository.interface';
+import { AppLoggerService } from '../../infra/logging/app-logger.service';
+import { BasicFolder } from '../../domain/entities/BasicFolder';
+import { Result } from 'rich-domain';
+import { differenceBy } from 'lodash';
+import { BasicFolderChange } from '../interfaces/basic-folder-changes.interface';
+import { RetryUtils } from 'src/utils/RetryUtils/RetryUtils';
+
+export interface BasicFolderChangeDetectionServiceOptions {
+    repository: RomachRepositoryInterface;
+    logger: AppLoggerService;
+    maxRetry: number;
+}
+
+export class BasicFolderChangeDetectionService {
+    constructor(
+        private readonly options: BasicFolderChangeDetectionServiceOptions
+    ) { }
+
+    async execute(current: BasicFolder[]): Promise<Result<BasicFolderChange>> {
+
+        const foldersIds = current.map(folder => folder.getProps().id)
+
+        const previousFoldersResult = await this.foldersServiceChanges(foldersIds);
+
+        if (previousFoldersResult.isFail()) {
+            return Result.fail()
+        }
+
+        const previousFoldersIdsAndUpdatedAt = previousFoldersResult.value();
+
+        const deleted = current.filter(folder => folder.getProps().deleted).map(folder => folder.getProps().id);
+
+        const updated = differenceBy(current, previousFoldersIdsAndUpdatedAt, 'id', 'updatedAt');
```

Comment:
I think this line may be incorrect. it compares BOTH id and updatedAt and it is not what you want. it will add the new folders to the updated